### ISSUE TITLE
Detached windows are kept in memory during spec run

### ISSUE THREAD

**Issue by mindtraveller** [cypress-io/cypress#29708]
### Test code to reproduce

We have a spec with 30+ tests, all the tests with number 30+ start failing due to memory issues `WebAssembly.instantiate(): Out of memory: Cannot allocate Wasm memory for new instance`. 
Research's shown that it might be related to the fact that the `Detached Window`s are kept in the memory. It can be easily reproduced by retrying a single test multiple times and then inspecting Memory snapshot. `Detached Window` instances in turn hold the pointers to the internal objects of our application preventing garbage collection.

Here I restarted test couple of times and the amount of `Detached Window`s grows.
<img width="446" alt="image" src="https://github.com/cypress-io/cypress/assets/11229620/dd8993fc-84ca-43af-bd2a-dbcfc7a2f5e3">

None of the mentioned in the template of this issue options help: 
- `experimentalMemoryManagement: true`
- `numTestsKeptInMemory: 0`

### Cypress Mode

cypress run

### Cypress Version

13.12.0

### Browser Version

Chrome 125

### Node version

v18.16.1

### Operating System

macOS 14.4.1

### Memory Debug Logs

```shell
cannot get them with the instructions provided
```

**Comment by jennifer-shehane** [comment]
To clarify, you're running via `cypress open` and clicking this retry button multiple times before inspecting the Memory snapshot?

**Comment by mindtraveller** [comment]
That's right, it's the easiest way to reproduce and debug the issue. Initially we faced the issue during `cypress run` on CI after we added 30nth test to a spec.
